### PR TITLE
JOS-106: Rename operation without testing code

### PR DESCRIPTION
**Pull request by shivanshu21**

### CODE REVIEW

File: src/rgw/rgw_common.h [hunk 1]
```diff
@@ -69,6 +69,7 @@ using ceph::crypto::MD5;
 #define RGW_ATTR_SHADOW_OBJ    	RGW_ATTR_PREFIX "shadow_name"
 #define RGW_ATTR_MANIFEST    	RGW_ATTR_PREFIX "manifest"
 #define RGW_ATTR_USER_MANIFEST  RGW_ATTR_PREFIX "user_manifest"
+#define RGW_ATTR_RENAME_MUTEX   RGW_ATTR_PREFIX "rename_mutex"
```
Comment (shivanshu21) [inline src/rgw/rgw_common.h]: not needed

File: src/rgw/rgw_op.cc [hunk 2]
```diff
@@ -1746,6 +1746,7 @@ void RGWPutObj::execute()
   int len;
   map<string, string>::iterator iter;
   bool multipart;
+  bool isRenameRunning = false;
```
Comment (shivanshu21) [inline src/rgw/rgw_op.cc]: remove

File: src/rgw/rgw_common.h [hunk 3]
```diff
@@ -146,6 +147,13 @@ using ceph::crypto::MD5;
 #define ERR_SIGNATURE_NO_MATCH   2027
 #define ERR_INVALID_ACCESS_KEY   2028
 #define ERR_BUCKET_ALREADY_OWNED 2029
+#define ERR_BAD_RENAME_REQ       2030
+#define ERR_RENAME_NOT_ENABLED   2031
+#define ERR_RENAME_FAILED        2032
+#define ERR_RENAME_DATA_LOST     2033
+#define ERR_RENAME_COPY_FAILED   2034
+#define ERR_RENAME_NEW_OBJ_DEL_FAILED     2035
+#define ERR_RENAME_OBJ_EXISTS    2037
```
Comment (shivanshu21) [inline src/rgw/rgw_common.h]: 2036

File: src/rgw/rgw_op.h [hunk 4]
```diff
@@ -25,6 +25,8 @@
 #include "rgw_cors.h"
 #include "rgw_quota.h"
 
+#define RENAME_OP_TESTING_FAULTS 1
```
Comment (shivanshu21) [inline src/rgw/rgw_op.h]: remove

File: src/rgw/rgw_rados.h [hunk 5]
```diff
@@ -1296,89 +1296,6 @@ class RGWRados
     zone_name = name;
   }
```
Comment (bafna) [inline src/rgw/rgw_rados.h]: Why this block is removed  ?

**Pull request merged**